storage/src/main/java/org/apache/kafka/storage/internals/log/LogOffsetMetadata.java

```diff
@@ -51,6 +57,8 @@ public LogOffsetMetadata(long messageOffset,
 
     // check if this offset is already on an older segment compared with the given offset
     public boolean onOlderSegment(LogOffsetMetadata that) {
+        if (this.segmentBaseOffset == REMOTE_LOG_UNKNOWN_OFFSET || that.segmentBaseOffset == REMOTE_LOG_UNKNOWN_OFFSET)
```

Comment:
We probably don't need this. If messageOffsetOnly() is true, we can just return false.